ext/pgsql/pgsql.c

```diff
@@ -3470,12 +3470,29 @@ static void php_pgsql_do_async(INTERNAL_FUNCTION_PARAMETERS, int entry_type)
 			PQconsumeInput(pgsql);
 			RETVAL_LONG(PQisBusy(pgsql));
 			break;
-		case PHP_PG_ASYNC_REQUEST_CANCEL:
-			RETVAL_LONG(PQrequestCancel(pgsql));
+		case PHP_PG_ASYNC_REQUEST_CANCEL: {
+			PGcancel *c;
+			char err[256];
+			int rc;
+
+			c = PQgetCancel(pgsql);
+
+			if (!c) {
+				// only possible reason since the connection is already verified at that point
+				zend_error(E_WARNING, "can't allocate a cancel procedure");
```

Comment:
Doesn't pgsql use ZMM? in which case the engine will bailout before.

If not, well that's something that is suboptimal but requires pgsql to allow setting a custom allocator.

PS: still on holiday for a week but going through the wooping amount of emails I got

Reply:
ah good point